components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -160,199 +205,137 @@ + (NSDictionary *)catalogMetadata {
-  bannerView.numberOfButtons = 2;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
 
-  UIFont *buttonFont = [MDCTypography body2Font];
-  MDCButton *dismissButton = bannerView.buttons[0];
+  MDCButton *dismissButton = [[MDCButton alloc] initWithFrame:CGRectZero];
   [dismissButton setTitle:@"DISMISS" forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [dismissButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
   [dismissButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   dismissButton.backgroundColor = self.colorScheme.surfaceColor;
   [dismissButton sizeToFit];
   [dismissButton addTarget:self
                     action:@selector(dismissBanner)
           forControlEvents:UIControlEventTouchUpInside];
-  MDCButton *changeTextButton = bannerView.buttons[1];
-  [changeTextButton setTitle:@"CHANGE TEXT" forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateNormal];
-  [changeTextButton setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  MDCButton *changeTextButton = [[MDCButton alloc] initWithFrame:CGRectZero];
+  [changeTextButton setTitle:@"LONG DISMISS" forState:UIControlStateNormal];
   [changeTextButton setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   changeTextButton.backgroundColor = self.colorScheme.surfaceColor;
   [changeTextButton sizeToFit];
   [changeTextButton addTarget:self
-                       action:@selector(changeText)
+                       action:@selector(dismissBanner)
              forControlEvents:UIControlEventTouchUpInside];
-
-  self.bannerView = bannerView;
-  [self.bannerViewContainer addSubview:bannerView];
-  CGSize bannerViewSize =
-      [bannerView sizeThatFits:CGSizeMake(exampleBannerContentMultiLineWidth, CGFLOAT_MAX)];
-  bannerView.frame = CGRectMake(0.0f, 0.0f, bannerViewSize.width, bannerViewSize.height);
+  NSMutableArray *buttons = [NSMutableArray arrayWithObjects:dismissButton, changeTextButton, nil];
+  bannerView.buttons = buttons;
 
   // Adjust bannerViewContainer's frame
   CGFloat topAreaInset = 0.0f;
   if (@available(iOS 11.0, *)) {
     topAreaInset = self.view.safeAreaInsets.top;
   }
-  self.bannerViewContainer.frame = CGRectMake(
-      0.0f, topAreaInset, [UIScreen mainScreen].bounds.size.width, bannerViewSize.height);
-  bannerView.center = CGPointMake(self.bannerViewContainer.frame.size.width / 2,
-                                  self.bannerViewContainer.frame.size.height / 2);
+  bannerView.frame = CGRectMake(0.0f, topAreaInset, CGRectGetWidth(bannerView.frame),
+                                CGRectGetHeight(bannerView.frame));
+}
+
+- (void)showMultiLineAlignedButtonStyleBannerWithIcon {
+  [self showMultiLineAlignedButtonStyleBanner];
+  self.bannerView.icon = [UIImage imageNamed:@"Email"];
```

Comment:
Same concern about using catalog-only assets in an Examples file. Please include the assets with the Examples directly.

Reply:
Addressed in 9d20047e48f2aa27ac1e5fbf246c564e83827f3b.